play-json/shared/src/main/scala/Json.scala

```diff
@@ -414,6 +419,29 @@ object Json extends JsonFacade {
   sealed trait MacroOptions
 
   /**
+   * Defines the default macro options.
+   *
+   * Since you can't have defaults for a type parameter (unless it's a contravariant type, it will default to
+   * Nothing), we supply the default via an implicit parameter.
+   */
+  trait DefaultMacroOptions[O <: Json.MacroOptions]
+
+  /**
+   * Low priority implicit used when Json.MacroOptions is not passed.
```

Comment:
It does take a `MacroOptions` type, and a type class of `DefaultMacroOptions`. See the other comments, this is the simplest way to implement default types for type parameters. I want the type returned by `apply` to default to `MacroOptions` when no type is explicitly passed, not `Nothing`.